modules/apps/bulk/bulk-rest-test/src/testIntegration/java/com/liferay/bulk/rest/resource/v1_0/test/BulkActionResourceTest.java

```diff
@@ -718,6 +720,138 @@ private void _testPostBulkActionWithTypeDefaultPermission()
 		Assert.assertTrue(jsonObject.has("test3"));
 	}
 
+	private void _testPostBulkActionWithTypeDefaultPermissionSingleRole()
+		throws Exception {
+
+		DefaultPermissionBulkAction defaultPermissionBulkAction =
+			new DefaultPermissionBulkAction();
+
+		Role role1 = RoleTestUtil.addRole(RoleConstants.TYPE_REGULAR);
+		Role role2 = RoleTestUtil.addRole(RoleConstants.TYPE_REGULAR);
+
+		_roleLocalService.addGroupRole(
+			testGroup.getGroupId(), role1.getRoleId());
+		_roleLocalService.addGroupRole(
+			testGroup.getGroupId(), role2.getRoleId());
+
+		ObjectDefinition objectDefinition =
+			ObjectDefinitionLocalServiceUtil.
+				getObjectDefinitionByExternalReferenceCode(
+					"L_CMS_DEFAULT_PERMISSION", _depotEntry2.getCompanyId());
+
+		ObjectEntryFolder objectEntryFolder1 =
+			_objectEntryFolderLocalService.
+				getObjectEntryFolderByExternalReferenceCode(
+					ObjectEntryFolderConstants.EXTERNAL_REFERENCE_CODE_CONTENTS,
+					_depotEntry2.getGroupId(), _depotEntry2.getCompanyId());
+
+		ObjectEntryFolder objectEntryFolder2 =
+			_objectEntryFolderLocalService.addObjectEntryFolder(
+				RandomTestUtil.randomString(), _depotEntry2.getGroupId(),
+				TestPropsValues.getUserId(),
+				objectEntryFolder1.getObjectEntryFolderId(), "",
+				HashMapBuilder.put(
+					LocaleUtil.ENGLISH, RandomTestUtil.randomString()
+				).build(),
+				RandomTestUtil.randomString(),
+				ServiceContextTestUtil.getServiceContext());
+
+		ObjectEntryFolder objectEntryFolder3 =
+			_objectEntryFolderLocalService.addObjectEntryFolder(
+				RandomTestUtil.randomString(), _depotEntry2.getGroupId(),
+				TestPropsValues.getUserId(),
+				objectEntryFolder1.getObjectEntryFolderId(), "",
+				HashMapBuilder.put(
+					LocaleUtil.ENGLISH, RandomTestUtil.randomString()
+				).build(),
+				RandomTestUtil.randomString(),
+				ServiceContextTestUtil.getServiceContext());
+
+		JSONObject defaultPermissionsJSONObject = JSONUtil.put(
+			ObjectEntryFolderConstants.EXTERNAL_REFERENCE_CODE_CONTENTS,
+			JSONUtil.put(
+				role1.getName(), JSONUtil.putAll(ActionKeys.VIEW)
+			).put(
+				role2.getName(), JSONUtil.putAll(ActionKeys.VIEW)
+			));
+
+		defaultPermissionBulkAction.setDefaultPermissions(
+			defaultPermissionsJSONObject.toString());
+
+		defaultPermissionBulkAction.setDepotGroupId(_depotEntry2.getGroupId());
+		defaultPermissionBulkAction.setBulkActionItems(
+			new BulkActionItem[] {
+				_toBulkActionItem(objectEntryFolder2),
+				_toBulkActionItem(objectEntryFolder3)
+			});
+
+		defaultPermissionBulkAction.setType(
+			BulkAction.Type.DEFAULT_PERMISSION_BULK_ACTION);
+
+		_postBulkAction(defaultPermissionBulkAction);
+
+		JSONObject singleRoleDefaultPermissionsJSONObject = JSONUtil.put(
+			ObjectEntryFolderConstants.EXTERNAL_REFERENCE_CODE_CONTENTS,
+			JSONUtil.put(
+				role1.getName(),
+				JSONUtil.putAll(ActionKeys.DELETE, ActionKeys.UPDATE)
+			).put(
+				role2.getName(), JSONUtil.putAll(ActionKeys.VIEW)
```

Comment:
in order to verify if we touch it or not the action here for the role2 should be different, and in the assert we verify that is has just the VIEW and not this new one